elements/rh-secondary-nav/rh-secondary-nav-menu.ts

```diff
@@ -0,0 +1,163 @@
+  private _sectionsNodes: NodeListOf<HTMLElement> | undefined;
+
+  @query('#sections') _sections: HTMLElement | undefined;
+
+  @query('#cta') _cta: HTMLElement | undefined;
+
+  @query('#container') _container: HTMLElement | undefined;
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    this.id ||= getRandomId('rh-secondary-nav-menu');
+  }
+
+  /**
+   * `isMobile` property is true when viewport `(min-width: ${tabletLandscapeBreakpoint})`.
+   * Property is observed for changes, and its value is updated using matchMediaControler
+   * when viewport changes at breakpoint or first load of the component.
+   */
+  @observed
+  @property({ reflect: false }) _isMobile = false;
+
+  /**
+   * Checks if passed in element is a RhSecondaryNavDropdown
+   * @param element:
+   * @returns {boolean}
+   */
+  static isDropdown(element: Element|null): element is RhSecondaryNavDropdown {
+    return element instanceof RhSecondaryNavDropdown;
+  }
+
+  /**
+   * Updates isMobile property with matchMedia when viewport changes
+   */
+  protected matchMedia = new MatchMediaController(this, `(min-width: ${tabletLandscapeBreakpoint})`, {
+    onChange: ({ matches }) => {
+      this._isMobile = !matches;
+    }
+  });
+
+  firstUpdated() {
+    this._isMobile = (window.outerWidth < parseInt(tabletLandscapeBreakpoint.toString().split('px')[0]));
+  }
+
+  render() {
+    const classes = { 'is-mobile': this._isMobile };
+
+    let menuContents;
+    if (this.type === 'full-width') {
+      menuContents = this.#fullWidthMenu();
+    } else {
+      menuContents = this.#fixedWidthMenu();
+    }
+    return html`<div id="container" class="${classMap(classes)}">${menuContents}</div>`;
+  }
+
+  /**
+   * Toggles visibility of menu by added or removing a `visible` attribute to the host
+   * @param visible {boolean}
+   */
+  toggleVisibility(visible: boolean) {
+    if (visible) {
+      this._container?.classList.add('visible');
+    } else {
+      this._container?.classList.remove('visible');
+    }
+  }
+
+  #fullWidthMenu() {
+    return html`
+      <div id="full-width" part="full-width">
+        <slot name="base">
+          <div id="sections" part="sections" hidden>
+            <slot name="section" @slotchange=${this.#onSectionsSlotChange}></slot>
+          </div>
+          <div id="cta" part="cta" hidden>
+            <slot name="cta" @slotchange=${this.#onCtaSlotChange}></slot>
+          </div>
+        </slot>
+      </div>
+    `;
+  }
+
+  #fixedWidthMenu() {
+    return html`
+      <div id="fixed-width" part="fixed-width">
+        <slot name="base">
+          <slot name="links"></slot>
+        </slot>
+      </div>
+    `;
+  }
+
+  /**
+   * When CTA slot is changed, check to see if it has children
+   * if it does then remove hidden attributes
+   * @returns {void}
+   */
+  #onCtaSlotChange(): void {
+    if (!this._ctaNodes) {
+      return;
+    }
+    const hasCta = this._ctaNodes.length > 0;
```

Comment:
```suggestion
    this._cta?.toggleAttribute('hidden', this._ctaNodes.length > 0);
```
https://developer.mozilla.org/en-US/docs/Web/API/Element/toggleAttribute#parameters

Reply:
you could also avoid the `if(p) return` by nullish coalescing to 0 with optional chaining on the `toggleAttribute` call

Reply:
but in any case, we should prefer to template this based on component state, rather than doing imperative updates

Reply:
Cleaned up a bit in f5c0101068cec480c8513addb9eb25fb85abe708.  However, I would like to talk about the "preferred" method more.  I might be missing some points on how we can use the component state as it relates to checking if the slot has slotted children.  If there is another method I would love to learn it.

Reply:
Fixed in 4e6a18a554cbf20c1268d594fdadca59cf0d9b94